ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/OngoingAuthorizationObserver.scala

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.ledger.api.auth
+
+import io.grpc.stub.{ServerCallStreamObserver, StreamObserver}
+
+final class OngoingAuthorizationObserver[A](
+    observer: StreamObserver[A],
+    claims: Claims,
+    authorized: Claims => Boolean,
+    throwOnFailure: => Throwable)
+    extends ServerCallStreamObserver[A] {
+
+  private[this] val wrapped = observer match {
```

Comment:
I'd move this check either to the apply method of the companion object, or simply right before this class is used. The implementation of this class can only work with `ServerCallStreamObserver` values, so it'd make sense to make this requirement public in the constructor.

Reply:
Absolutely right, thanks for the comment. :slightly_smiling_face:

Reply:
Addressed by https://github.com/digital-asset/daml/pull/3286/commits/20a4635add5f912c6e6c5fd78e06de61846a1171